calico/felix/fetcd.py

```diff
@@ -334,6 +335,54 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self.estimated_host_count()
+        self._usage_count_greenlet = gevent.Greenlet(
+            self._periodically_usage_count
+        )
+        # self._usage_count_greenlet.link_exception(self._on_worker_died)
+
+    # def _on_worker_died(self, watch_greenlet):
+    #     """
+    #     Greenlet: spawned by the gevent Hub if the etcd watch loop ever
+    #     stops, kills the process.
+    #     """
+    #     _log.critical("Worker greenlet died: %s; exiting.", watch_greenlet)
+    #     sys.exit(1)
+
+    def estimated_host_count(self):
+        return len(self.ipv4_by_hostname)
```

Comment:
Please add a docstring explaining what this returns and that it's safe to call from another thread.